api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -888,21 +899,30 @@ type AzureNodePoolPlatform struct {
 	// +kubebuilder:validation:Enum=Standard_LRS;StandardSSD_LRS;Premium_LRS;UltraSSD_LRS
 	// +optional
 	DiskStorageAccountType string `json:"diskStorageAccountType,omitempty"`
-	// AvailabilityZone of the nodepool. Must not be specified for clusters
-	// in a location that does not support AvailabilityZone.
+
+	// AvailabilityZone is the failure domain identifier where the VM should be attached to. This must not be specified
+	// for clusters in a location that does not support AvailabilityZone.
+	//
 	// +optional
 	AvailabilityZone string `json:"availabilityZone,omitempty"`
+
 	// DiskEncryptionSetID is the ID of the DiskEncryptionSet resource to use to encrypt the OS disks for the VMs.
+	//
 	// +optional
 	DiskEncryptionSetID string `json:"diskEncryptionSetID,omitempty"`
```

Comment:
Similar comment as https://github.com/openshift/hypershift/pull/4004/files#r1595143214

Reply:
Updated

Reply:
From this doc I understand the resource group can be a different one than in the hosted cluster right? I am asking this because the disk encryption set will be provided by the end-user

Reply:
Yeah, the one in the Hosted Cluster will be the managed service resource group